src/components/CheckEditor/CheckEditorNew.test.tsx

```diff
@@ -257,4 +257,50 @@ describe('new checks', () => {
       },
     });
   });
+
+  describe(`over limits`, () => {
+    it(`shows error alert when check limit is reached`, async () => {
+      server.use(
+        apiRoute('getTenantLimits', {
+          result: () => ({
+            json: {
+              MaxChecks: 1,
+              MaxScriptedChecks: 10,
+            },
+          }),
+        })
+      );
+      render(<CheckForm />, {
+        route: `${PLUGIN_URL_PATH}${ROUTES.Checks}/new/:checkType`,
+        path: `${PLUGIN_URL_PATH}${ROUTES.Checks}/new/http`,
+      });
+
+      const errorAlert = await screen.findByText(/Maximum number of checks reached/, { exact: false });
+      expect(errorAlert).toBeInTheDocument();
+      const submitButton = await screen.findByRole('button', { name: 'Save' });
+      expect(submitButton).toBeDisabled();
+    });
+
+    it(`shows error alert when scripted check limit is reached`, async () => {
+      server.use(
+        apiRoute('getTenantLimits', {
+          result: () => ({
+            json: {
+              MaxChecks: 10,
+              MaxScriptedChecks: 1,
+            },
+          }),
+        })
+      );
+      render(<CheckForm />, {
+        route: `${PLUGIN_URL_PATH}${ROUTES.Checks}/new/:checkType`,
+        path: `${PLUGIN_URL_PATH}${ROUTES.Checks}/new/scripted`,
+      });
+
+      const errorAlert = await screen.findByText(/Maximum number of scripted checks reached/, { exact: false });
+      expect(errorAlert).toBeInTheDocument();
+      const submitButton = await screen.findByRole('button', { name: 'Save' });
```

Comment:
I know we just got rid of the `byRole` selectors, but `byText` returns the span inside the button, so I can't assert whether it's disabled or not

Reply:
I'm happy to let them sneak back in and if we start seeing timeouts reappear we can re-address this. We have some helper functions so there is an avenue to abstract some of this (I think how I handled this in other tests is I got the parentNode from the getByText selector... 🤮)